src/backend/columnar/columnar_metadata.c

```diff
@@ -1276,6 +1276,8 @@ BuildStripeMetadata(Relation columnarStripes, HeapTuple heapTuple)
 
 	TransactionId entryXmin = HeapTupleHeaderGetXmin(heapTuple->t_data);
 	stripeMetadata->aborted = TransactionIdDidAbort(entryXmin);
+	stripeMetadata->insertedByCurrentXact =
+		TransactionIdIsCurrentTransactionId(entryXmin);
```

Comment:
Not sure, but we anyway throw an error if there are pending writes in the upper transaction.
Added a test for that.

Reply:
Might be worth a comment that we handle both cases:

1. If there is unflushed data in a parent xact, then we would have already thrown an error
2. If the data is from an earlier subxact that committed, then it would have been flushed